### PR TITLE
Move OpenTelemetry.AutoInstrumentation.Native.so / OpenTelemetry.AutoInstrumentation.Native.dylib to a directory in tracer-home

### LINKED ISSUE #1866: Move OpenTelemetry.AutoInstrumentation.Native.so / OpenTelemetry.AutoInstrumentation.Native.dylib to a directory
Currently, `OpenTelemetry.AutoInstrumentation.Native.dylib` and `OpenTelemetry.AutoInstrumentation.Native.so` are placed in `tracer-home`. Proposal is to move this to a new directory in `tracer-home`.

For example, we could create a folder named `misl` and place `OpenTelemetry.AutoInstrumentation.Native.so` in it.  Conceptually this will make it easier to pack with Nuget.

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
## Why

Fixes #1866 

## What

Used existing environment variable `OS_TYPE` to decide the directory name. For alpine to produce a directory name, set the environment variable in pipeline. All other platforms falls back to default directory name. 

Follow up: Create an issue to update Nuget with alpine native library. 

## Tests

- Current tests uses the native library from new path

## Checklist

- [X] `CHANGELOG.md` is updated.
- [X] Documentation is updated.
- [X] New features are covered by tests.

### CODE REVIEW

File: dev/envvars.sh [hunk 1]
```diff
@@ -1,5 +1,18 @@
 #!/bin/bash
 
+case "$(uname -s | tr '[:upper:]' '[:lower:]')" in
+  linux*)
+    if [ "$(ldd /bin/ls | grep -m1 'musl')" ]; then
+      CLRDIRNAME="linux-musl"
```
Comment (pjanotti) [inline dev/envvars.sh:6]: Small naming suggestion to make clear where the values come from
```suggestion
      DOTNET_RUNTIME_ID="linux-musl"
```

File: instrument.sh [hunk 2]
```diff
@@ -21,7 +21,16 @@ fi
 
 # validate input
 case "$OS_TYPE" in
-  "linux-glibc"|"linux-musl"|"macos"|"windows")
+  "linux-glibc")
+    CLRDIRNAME="linux-x64"
```
Comment (pjanotti) [inline instrument.sh:25]: Same rename suggestion as above.

**Review by pjanotti — APPROVED** [review]
LGTM, just one variable naming suggestion

**Review by pjanotti — APPROVED** [review]

File: dev/envvars.sh [hunk 3]
```diff
@@ -1,5 +1,18 @@
 #!/bin/bash
 
+case "$(uname -s | tr '[:upper:]' '[:lower:]')" in
+  linux*)
+    if [ "$(ldd /bin/ls | grep -m1 'musl')" ]; then
```
Comment (zacharycmontoya) [inline dev/envvars.sh:5]: I haven't run this command myself but if you say this works I'll take your word for it 😄

**Review by zacharycmontoya — APPROVED** [review]
LGTM, nice work

File: dev/envvars.sh (hunk 3 above)
Comment (rajkumar-rangaraj) [reply dev/envvars.sh:5]: This is a copy-paste code from [instrument.sh](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/c7b279bcdf3510e17e6d32707b5ba919f60488ed/instrument.sh#L10) which works as expected. I think @pellared wrote the logic for it and tested out.

**Pull request merged**